### ISSUE TITLE
Potential bug on setting trainable layers for each stage

### ISSUE THREAD

**Issue by hli2020** [multimodallearning/pytorch-mask-rcnn#20]
https://github.com/multimodallearning/pytorch-mask-rcnn/blob/809abba590db89779ac02c42286135f18ea08b53/model.py#L1493

C1-C4 will always be set to `requires_grad=False` even after stage `heads` is passed onto `4+` or `all`.

**Comment by lasseha** [comment]
For `heads` this is the expected behavior, isn't it? For `4+` only C1-C3 do not require gradient and for `all` i just checked that for no parameter `requires_grad` is set to `false`, so i do not see a bug here.

**Comment by hli2020** [comment]
@lasseha 
Do you directly run the last stage `all`? in this case then all layers have `requires_grad = True`.

If you run first stage, C1-C5 are all `requires_grad=False`; then you enter the second stage, `4+`, `not_trainable list` is from C1-C3, it sets these layers (C1-C3) to `requires_grad=False`. However, the flag of C4, C5 **doesn't** change; they are still `requires_grad=False`. So after `if xxx`, there should be an `else: param[1].requires_grad = True`. See [here](https://github.com/hli2020/pytorch-mask-rcnn/blob/master/lib/model.py#L131).
